src/crypto/hash.c

```diff
@@ -48,3 +48,8 @@ void cn_fast_hash(const void *data, size_t length, char *hash) {
   hash_process(&state, data, length);
   memcpy(hash, &state, HASH_SIZE);
 }
+
+void cn_pad_by_fast_hash__C(const uint8_t *in, size_t inlen, uint8_t *md, int mdlen)
+{
+  keccak(in, inlen, md, mdlen);
+}
```

Comment:
Hmmm I'm not sure. I would say not in this case because the rest of the file does not do so but I could see why new code might want to do it.

Reply:
> because the rest of the file does not do so

I'm not sure what you mean by this. This file, as well as probably all the source code in the codebase, ends with an empty line.

Reply:
Oh ok, I misinterpreted where the newline you were referring to would be. Thought you meant somewhere else.

Reply:
This is done